packages/react-instantsearch-core/src/connectors/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,220 @@
+import algoliasearchHelper, {
+  PlainSearchParameters,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import createConnector, { ConnectedProps } from '../core/createConnector';
+import {
+  omit,
+  getObjectType,
+  removeEmptyKey,
+  removeEmptyArraysFromObject,
+} from '../core/utils';
+import {
+  refineValue,
+  getIndexId,
+  hasMultipleIndices,
+} from '../core/indexUtils';
+
+type Hit = any;
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+interface ConfigureRelatedItemsProps {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: Hit;
+  /**
+   * The schema to create the optional filters.
+   * Each key represents an attribute from the hit.
+   */
+  matchingPatterns: MatchingPatterns;
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+function createOptionalFilter({
+  attributeName,
+  attributeValue,
+  attributeScore,
+}) {
+  return `${attributeName}:${attributeValue}<score=${attributeScore || 1}>`;
```

Comment:
probably the same for InstantSearch.js, but don't we need to filter out & escape some values?

Reply:
Filtering out based on what? Only `string | string[]` are taken into account.

Regarding escaping, yes, we probably should. Any suggestions as to how escaping the values for Algolia to understand them all the time?

Reply:
a value can not start with `- `, since that's a negation, quotes need to be escaped, since they have special meaning to group things, and `<`, `>` needs to be escaped because it's xml and used in score (escaped means that in the final json there's `\\<`, because the backslash needs to be escaped itself.

I can't find a reference document for that

Reply:
https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/#usage-notes, I was correct according to @xroche, but `<` & `>` need to be removed, not escaped

Reply:
Thanks for looking into it! I'll take care of that in a stacked PR.

Reply:
Done in #2884.